### PR TITLE
feat: Add V3 API for process step results

### PR DESCRIPTION
**Pull request by BjarkeMeier**
## Description

Initial work on the new V3 API for Wholesale.

## References

* #782

### CODE REVIEW

File: source/dotnet/Services/WebApi/V3/ProcessStepResult/ProcessStepResultController.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        _processStepApplicationService = processStepApplicationService;
+        _processStepResultFactory = processStepResultFactory;
+        _batchApplicationService = batchApplicationService;
+    }
+
+    [AllowAnonymous] // TODO: Temporary hack to enable EDI integration while awaiting architects decision
+    [HttpGet]
+    public async Task<ProcessStepResultDto> GetAsync([FromRoute] Guid batchId, string gridAreaCode, TimeSeriesType timeSeriesType, string gln)
+    {
+        var request = new ProcessStepResultRequestDtoV2(batchId, gridAreaCode, timeSeriesType, gln);
```
Comment (lasrinnil) [inline source/dotnet/Services/WebApi/V3/ProcessStepResult/ProcessStepResultController.cs:45]: shold we keep the dto?

File: source/dotnet/Services/WebApi/V3/ProcessStepResult/ProcessStepResultFactory.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    {
+        return new ProcessStepResultDto(
+            stepResult.Sum,
+            stepResult.Min,
+            stepResult.Max,
+            batch.PeriodStart,
+            batch.PeriodEnd,
+            batch.Resolution,
+            batch.Unit,
+            stepResult.TimeSeriesPoints.Select(Map()).ToArray());
```
Comment (lasrinnil) [inline source/dotnet/Services/WebApi/V3/ProcessStepResult/ProcessStepResultFactory.cs:31]: MapMap

File: source/dotnet/Services/WebApi/V3/ProcessStepResult/TimeSeriesPointQuality.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Energinet.DataHub.Wholesale.WebApi.V3.ProcessStepResult;
+
+public static class TimeSeriesPointQuality
```
Comment (lasrinnil) [inline source/dotnet/Services/WebApi/V3/ProcessStepResult/TimeSeriesPointQuality.cs:17]: Enum?

File: source/dotnet/Services/WebApi/V3/ProcessStepResult/ProcessStepResultFactory.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Energinet.DataHub.Wholesale.Application.Batches.Model;
+
+namespace Energinet.DataHub.Wholesale.WebApi.V3.ProcessStepResult;
+
+public class ProcessStepResultFactory : IProcessStepResultFactory
+{
+    public ProcessStepResultDto Create(Contracts.ProcessStepResultDto stepResult, BatchDto batch)
```
Comment (lasrinnil) [inline source/dotnet/Services/WebApi/V3/ProcessStepResult/ProcessStepResultFactory.cs:21]: Contracts.ProcessStepResultDto stepResult still relevant?

File: source/dotnet/Services/WebApi/Configuration/ServiceCollectionExtensions.cs [hunk 5]
```diff
@@ -63,6 +64,11 @@ public static void AddJwtTokenSecurity(this IServiceCollection serviceCollection
         serviceCollection.AddPermissionAuthorization();
     }
 
+    public static void AddWebApiHostRegistrations(this IServiceCollection serviceCollection)
```
Comment (lasrinnil) [inline source/dotnet/Services/WebApi/Configuration/ServiceCollectionExtensions.cs:67]: Is this not a part of the "command" stack?

File: source/dotnet/Services/WebApi/V3/ProcessStepResult/ProcessStepResultController.cs (hunk 1 above)
Comment (BjarkeMeier) [reply source/dotnet/Services/WebApi/V3/ProcessStepResult/ProcessStepResultController.cs:45]: No 👍 Will refactor this clean up in an upcoming PR.

File: source/dotnet/Services/WebApi/V3/ProcessStepResult/TimeSeriesPointQuality.cs (hunk 3 above)
Comment (BjarkeMeier) [reply source/dotnet/Services/WebApi/V3/ProcessStepResult/TimeSeriesPointQuality.cs]: Fixed

File: source/dotnet/Services/WebApi/Configuration/ServiceCollectionExtensions.cs (hunk 5 above)
Comment (BjarkeMeier) [reply source/dotnet/Services/WebApi/Configuration/ServiceCollectionExtensions.cs]: Ok, updated.

File: source/dotnet/Services/WebApi.IntegrationTests/WebApi/V3/ProcessStepResultTests.cs [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        request.SetPrivateProperty(r => r.Type, TimeSeriesType.Production);
+        result.SetPrivateProperty(r => r.TimeSeriesPoints, new TimeSeriesPointDto[] { new(DateTimeOffset.Now, decimal.One, "A04") });
+        processStepApplicationServiceMock
+            .Setup(service => service.GetResultAsync(It.IsAny<ProcessStepResultRequestDto>()))
+            .ReturnsAsync(() => result);
+        batchApplicationServiceMock.Setup(service => service.GetAsync(request.BatchId)).ReturnsAsync(batchDto);
+        Factory.ProcessStepApplicationServiceMock = processStepApplicationServiceMock;
+        Factory.BatchApplicationServiceMock = batchApplicationServiceMock;
+
+        var expectedUrl = $"/v3/batches/{request.BatchId}/processes/{request.GridAreaCode}/time-series-types/{request.Type}";
```
Comment (JesperGomes) [inline source/dotnet/Services/WebApi.IntegrationTests/WebApi/V3/ProcessStepResultTests.cs:62]: or just 'url'?

File: source/dotnet/Services/WebApi/V3/ProcessStepEnergySupplier/ProcessStepEnergySupplierController.cs [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    /// <summary>
+    /// Energy suppliers associated with the balance responsible party specified by the <paramref name="balanceResponsibleParty"/>.
+    /// </summary>
+    [AllowAnonymous] // TODO: Temporary hack to enable EDI integration while awaiting architects decision
+    [HttpGet]
+    public async Task<List<ActorDto>> GetAsync([FromRoute] Guid batchId, [FromRoute] string gridAreaCode, [FromRoute] TimeSeriesType timeSeriesType, [FromQuery] string? balanceResponsibleParty)
+    {
+        if (balanceResponsibleParty == null)
```
Comment (JesperGomes) [inline source/dotnet/Services/WebApi/V3/ProcessStepEnergySupplier/ProcessStepEnergySupplierController.cs:55]: Not sure I understand the purpose here. A comment (or ToDo) would be nice

File: source/dotnet/Services/WebApi/V3/ProcessStepResult/ProcessStepResultController.cs [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+        _processStepResultFactory = processStepResultFactory;
+        _batchApplicationService = batchApplicationService;
+    }
+
+    /// <summary>
+    /// Get calculation result for a specific actor by GLN.
+    /// </summary>
+    [AllowAnonymous] // TODO: Temporary hack to enable EDI integration while awaiting architects decision
+    [HttpGet]
+    public async Task<ProcessStepResultDto> GetForActorAsync(
```
Comment (JesperGomes) [inline source/dotnet/Services/WebApi/V3/ProcessStepResult/ProcessStepResultController.cs:49]: Why not use overloads and then avoid null handling? One method could have[HttpGet]  and the other could have [HttpGet("{gln}")]?

**Review by JesperGomes — APPROVED** [review]

File: source/dotnet/Services/WebApi/V3/ProcessStepEnergySupplier/ProcessStepEnergySupplierController.cs (hunk 7 above)
Comment (lasrinnil) [reply source/dotnet/Services/WebApi/V3/ProcessStepEnergySupplier/ProcessStepEnergySupplierController.cs:55]: I'll leave a TODO here.

File: source/dotnet/Services/WebApi/V3/ProcessStepResult/ProcessStepResultController.cs (hunk 8 above)
Comment (lasrinnil) [reply source/dotnet/Services/WebApi/V3/ProcessStepResult/ProcessStepResultController.cs:49]: Please check my refactor :)

**Pull request merged**